_posts/2022-11-03-markdown-asciidoc-restructuredtext.md

```diff
@@ -0,0 +1,193 @@
+
+### reStructuredText Framework
+
+If you've noticed the usage "framework" instead of "frameworks" in the above heading, you can guess that there's only one framework for reStructuredText. [Sphinx](https://www.sphinx-doc.org/en/master/index.html),  originally created for Python documentation, takes plain-text files in reStructuredText format and transformes it into HTML, PDF, and any output formats. A few well-known projects that use Sphinx for documentation are [Django](https://docs.djangoproject.com/) and [ReadTheDocs](https://docs.readthedocs.io/en/latest/index.html). 
+
+## Choose the right markup
+
+If you're a developer, learning the syntax and styling for a markup language is relatively easy compared to a programming language. When you learn the syntax of a markup language and can render an HTML page locally, you reach the `hello world` nirvana of documentation. However, the difficulty doesn't necessarily start when you write the first documentation file. I [asked on Twitter](https://twitter.com/DewanAhmed/status/1583465169006448640?s=20&t=kOdJ_vOoZiZ_TJK0vJDK9A) about the choice among Markdown, AsciiDoc, and reStructuredText and this section will summarize the findings.
+
+### Markdown - simple and easy to get started
+
+#### Flavors
+
+Yes, there are [A LOT](https://github.com/commonmark/commonmark-spec/wiki/markdown-flavors) of flavors. Despite all eleventy-zillion flavors, Markdown is loved for developer documentation due to its simplicity. The argument against all the unknown Markdown flavors and incompatibility doesn't stand that much since most projects would choose a common flavor like GFM. 
+
+#### Purpose
+
+For many projects, documentation is about a solid README and some files under the `docs` folder on GitHub. They don't need macros or custom tags within their documentation. Since GitHub renders multiple flavors of Markdown nicely, Markdown is an easy choice for many projects.
+
+#### Syntax
+
+Let's look at a few syntax comparisons to understand how simplicity helps Markdown win many projects. 
+
+Link in reStructuredText:
+
+```
+This is `Dewan's blog <https://www.dewanahmed.com>`_.
+```
+
+Link in Asciidoc:
+
+```
+https://www.dewanahmed.com[Dewan's blog]
+```
+
+Link in Markdown:
+
+```
+[This is `Dewan's blog](https://www.dewanahmed.com)
+```
+
+Let's take a look at header complexity in reStructuredText: 
+
+```
+This is a header
+~~~~~~~~~~~~~~~~
+```
+
+```
+======================
+This is also a header
+======================
+```
+
+```
+~~~~~~~~~~
+And yes... this is also a header
+~~~~~~~~~~
+```
```

Comment:
No, this is not a header - the number of `~` characters is not equal to or more than the number of title characters.

If you want to make your case a little stronger, the valid characters for headers are all of
```
! " # $ % & ' ( ) * + , - . / : ; < = > ? @ [ \ ] ^ _ ` { | } ~
```
so choose whichever you think looks daftest 😄 (clearly the *recommended* set is shorter)

(I could explain why there's no pre-defined hierarchy of header characters, but you probably don't care, and absent the history, it does indeed look mad. I could also point out that I don't count them out by hand, because I use a decent text editor. but I'm aware that the support for rST in other editors is not as good as it should be 😢)

Reply:
Initially, I thought that I needed equal of more number of title characters. But then I saw this using an online editor:

<img width="1550" alt="Screen Shot 2022-11-02 at 11 43 42 PM" src="https://user-images.githubusercontent.com/30604461/199638398-2642b616-6731-4f81-8fbc-22a0538c5b47.png">

Is it a Sphinx thing then?